### PR TITLE
Added mamp4 versioned cask

### PR DESCRIPTION
**Pull request by daften**
After making all changes to the cask:

- [ X ] `brew cask audit --download {{cask_file}}` is error-free.
- [ X ] `brew cask style --fix {{cask_file}}` reports no offenses.
- [ X ] The commit message includes the cask’s name and version.
- [ X ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [ X ] Named the cask according to the [token reference].
- [ X ] `brew cask install {{cask_file}}` worked successfully.
- [ X ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ X ] Checked there are no [open pull requests] for the same cask.
- [ X ] Checked the cask was not [already refused].
- [ X ] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask-versions/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by daften** [comment]
The build fails on onuninstall, but that's because mamp uninstall simply sucks, nothing i can do in this PR about that.

File: Casks/mamp4.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on macos: '>= :yosemite'
+
+  pkg "MAMP_MAMP_PRO_#{version}.pkg"
+
+  postflight do
+    set_ownership ['/Applications/MAMP', '/Applications/MAMP PRO']
+  end
+
+  uninstall pkgutil: 'de.appsolute.installer.(mamp|mampacticon|mampendinstall|mamppro).pkg'
```
Comment (vitorgalvao) [inline Casks/mamp4.rb]: `pkgutil` takes wildcards, not regular expressions.

Comment (daften) [reply Casks/mamp4.rb]: this is a copy/paste from a working cask from homebrew-cask itself, so I'd think this should work. See https://github.com/commitay/homebrew-cask/blob/a4192f3bb1bc46ba6175e4234336524131073b44/Casks/mamp.rb

But mamp*.pkg should work?

Comment (daften) [reply Casks/mamp4.rb]: I think the package list is created based on a regex (pkgutill --pkgs, see http://www.manpagez.com/man/1/pkgutil/)

https://github.com/Homebrew/brew/blob/459e2bfa48ebed4bf49d6aebb68df1b416e9b0f3/Library/Homebrew/cask/lib/hbc/artifact/abstract_uninstall.rb#L211

Comment (vitorgalvao) [reply Casks/mamp4.rb]: Seems like we already changed it to support regexp; it’s even updated in the documentation. I remember thinking about it (I think I was even the one to open the issue) but I don’t recall it being implemented. Maybe I need a break.

**Pull request merged**